src/query_plan/mod.rs

```diff
@@ -1 +1,210 @@
+    /// useful  in practice, but not disallowed by the @defer spec at the moment).
+    sub_selection: Option<SelectionSet>,
+    /// The plan to get all the data for the primary block. Same notes as for subselection: usually
+    /// defined, but can be undefined in some corner cases where nothing is to be done in the
+    /// primary block.
+    node: Option<PlanNode>,
+}
+
+/// A deferred block of a `DeferNode`.
+pub struct DeferredDeferBlock {
+    /// References one or more fetch node(s) (by `id`) within `DeferNode.primary.node`. The plan of
+    /// this deferred part should not be started until all such fetches return.
+    depends: Vec<DeferredDependency>,
+    /// The optional defer label.
+    label: Option<NodeStr>,
+    /// Path, in the query, to the `@defer` application this corresponds to. The `sub_selection`
+    /// starts at this `query_path`.
+    query_path: Vec<QueryPathElement>,
+    /// The part of the original query that "selects" the data to send in the deferred response
+    /// (once the plan in `node` completes). Will be set _unless_ `node` is a `DeferNode` itself.
+    sub_selection: Option<SelectionSet>,
+    /// The plan to get all the data for this deferred block. Usually set, but can be `None` for a
+    /// `@defer` application where everything has been fetched in the "primary block" (i.e. when
+    /// this deferred block only exists to expose what should be send to the upstream client in a
+    /// deferred response), but without declaring additional fetches. This happens for @defer
+    /// applications that cannot be handled through the query planner and where the defer cannot be
+    /// passed through to the subgraph).
+    node: Option<PlanNode>,
+}
+
+pub struct DeferredDependency {
+    /// A `FetchNode` ID.
+    id: NodeStr,
+    /// If the `FetchNode` pointed to by `id` has `has_defers` as `true`, and this field is set (to
+    /// the label of one of the defers of said `FetchNode`), then this dependency is strictly for
+    /// the defers with that label.
+    defer_label: Option<NodeStr>,
+}
+
+pub struct ConditionNode {
+    condition: NodeStr,
+    if_clause: Option<PlanNode>,
+    else_clause: Option<PlanNode>,
+}
+
+/// The type of rewrites currently supported on the input/output data of fetches.
+///
+/// A rewrite usually identifies some sub-part of the data and some action to perform on that
+/// sub-part.
+pub enum FetchDataRewrite {
+    ValueSetter(FetchDataValueSetter),
+    KeyRenamer(FetchDataKeyRenamer),
+}
+
+/// A rewrite that sets a value at the provided path of the data it is applied to.
+pub struct FetchDataValueSetter {
+    /// Path to the value that is set by this "rewrite".
+    path: Vec<FetchDataPathElement>,
+    /// The value to set at `path`. Note that the query planner currently only uses string values,
+    /// but that may change in the future.
+    set_value_to: serde_json::value::Value,
```

Comment:
```suggestion
    set_value_to: serde_json::Value,
```
Ultra nitpick but generally using the top-level reexport is preferred

Reply:
Good catch!